app/models/post.rb

```diff
@@ -1,4 +1,13 @@
 class Post < ApplicationRecord
-  belongs_to :user, counter_cache: :posts_counter
-  has_many :likes, :comments
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  def update_posts_count
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [SUGGESTION] We can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the `update_posts_count` method every time a new post has been saved to the database, then make the `update_post_counter` method private.

  Something like this:
  
  ```ruby
  # app/models/post.rb
    after_save :update_posts_count
  
    private
  
    def update_posts_count
      author.increment!(:posts_counter)
    end
  ```